modules/vaos/spec/request/appointments_request_spec.rb

```diff
@@ -339,38 +340,24 @@
         end
 
         it 'returns bad request with detail in errors' do
-          VCR.use_cassette('vaos/appointments/put_cancel_appointment_400', match_requests_on: %i[method uri]) do
-            expect(Rails.logger).to receive(:warn).with('VAOS service call failed!', any_args)
+          VCR.use_cassette('vaos/appointments/put_cancel_appointment_409', match_requests_on: %i[method uri]) do
+            expect(Rails.logger).to receive(:warn).with('VAOS service call failed!', any_args).once
             expect(Rails.logger).to receive(:warn).with(
               'Clinic does not support VAOS appointment cancel',
               clinic_id: request_body[:clinic_id],
               site_code: request_body[:facility_id]
-            )
+            ).once
+            # We're checking that the logger is called twice to account for behavior in sentry_logging.rb lines 25-30
```

Comment:
Someone on VSP should be made aware of this. There is a lot of redundant logging happening there in SentryLogging as a result of that issue. Did someone really introduce an else clause there just to get "punctuation" into the logs? 🤔 

@erluti

Reply:
Looking at that code it looks like `Common::Exceptions::BackendServiceException` is being treated as more fully featured.  Although line 30 would add the backtrace a second time.

i'm not clear on how this code comment relates to that, though?

@kfrz i see you put 👀 here, do you have any insight?

Reply:
We put this code comment in here because once the changes to SentryLogging are made to make it not log the same thing twice for no good reason, this spec would be broken -- the thought was that whomever makes those fixes would see the comment and feel safe enough to fix the spec to be `.once` instead of `.twice` without further inquiry.

Reply:
We're talking about this code?
```
if exception.is_a? Common::Exceptions::BackendServiceException
      rails_logger(level, exception.message, exception.errors, exception.backtrace)
    else
      rails_logger(level, "#{exception.message}.")
    end
    rails_logger(level, exception.backtrace.join("\n")) unless exception.backtrace.nil?
 ```
 Looks like line 30 logs the backtrace a second time if it is a `Common::Exceptions::BackendServiceException` and the exception has errors and a backtrace.  I see your concern now.

Reply:
Ticket https://app.zenhub.com/workspaces/vsp-5cedc9cce6e3335dc5a49fc4/issues/department-of-veterans-affairs/va.gov-team/18076